### PR TITLE
v7-contentshare: refactor(plugin-conversation): Fixes issue with content share, downlo…

### PR DESCRIPTION
**Pull request by saurjai3**
Fixes issue with content share, download/upload/preview/progress

### CODE REVIEW

File: packages/plugin-conversation/src/conversation.js [hunk 1]
```diff
@@ -471,7 +471,13 @@ const Conversation = SparkPlugin.extend({
    * @param {Object} conversation
    * @returns {ShareActivty}
    */
-  makeShare(conversation) {
+  makeShare(conversation, activity) {
+    // if we pass activity as null then it does not take care of the clientTempId created by the web-client while making the provisional activity, hence we need to pass the activity which was created by the web-client
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/conversation.js]: wrap comments at 80 columns

File: packages/plugin-conversation/src/decryption-transforms.js [hunk 2]
```diff
@@ -27,6 +27,12 @@ export const transforms = toArray(`inbound`, {
       key = object.encryptionKeyUrl;
     }
 
+    // Fixed issue with transcoded content
+    if (object.objectType === 'transcodedContent') {
+      console.log('@@@@@ temporarily calling decryptFile directly from here until a fix is available.');
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/decryption-transforms.js]: remove `console.log`

File: packages/plugin-conversation/src/decryption-transforms.js [hunk 3]
```diff
@@ -27,6 +27,12 @@ export const transforms = toArray(`inbound`, {
       key = object.encryptionKeyUrl;
     }
 
+    // Fixed issue with transcoded content
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/decryption-transforms.js]: Please explain what issue is being fixed.

File: packages/plugin-conversation/src/share-activity.js [hunk 4]
```diff
@@ -92,18 +109,21 @@ const ShareActivity = SparkPlugin.extend({
...
     }, pick(options, `actions`));
 
     this.uploads.set(file, upload);
     const promise = this.detect(file)
       .then((type) => {
         upload.mimeType = type;
-        if (!file.type) {
-          file.type = type;
-        }
+        // The blow code gives an error that the file.type is a readonly property, hence refactoring the code.
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/share-activity.js]: wrap comments at 80, remove commented code

File: packages/plugin-conversation/src/share-activity.js [hunk 5]
```diff
@@ -176,6 +196,17 @@ const ShareActivity = SparkPlugin.extend({
       });
   },
 
+/**
```
Comment (ianwremmel) [inline packages/plugin-conversation/src/share-activity.js]: Fix indentation

File: packages/plugin-encryption/src/encryption.js [hunk 6]
```diff
@@ -34,7 +34,7 @@ const Encryption = SparkPlugin.extend({
 
         return scr.decrypt(b);
       })
-      .then(ensureBuffer);
+      .then((res) => res);
```
Comment (ianwremmel) [inline packages/plugin-encryption/src/encryption.js]: this doesn't do anything; remove the line entirely.

**Comment by saurjai3** [comment]
created a new PR: https://github.com/ciscospark/spark-js-sdk/pull/546
hence closing this one,

**Pull request closed without merging**